ui/app/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -99,24 +100,23 @@ export default class PermissionConnect extends Component {
   }
 
   redirectFlow (accepted) {
-    const { history } = this.props
+    const { history, location, confirmPermissionPath } = this.props
 
     this.setState({
       redirecting: true,
       permissionAccepted: accepted,
     })
     this.removeBeforeUnload()
 
-    if (getEnvironmentType() === ENVIRONMENT_TYPE_FULLSCREEN) {
-      setTimeout(async () => {
-        const currentTab = await global.platform.currentTab()
-        global.platform.closeTab(currentTab.id)
-      }, 2000)
-    } else if (getEnvironmentType() === ENVIRONMENT_TYPE_NOTIFICATION) {
+    if (getEnvironmentType() === ENVIRONMENT_TYPE_NOTIFICATION) {
       setTimeout(async () => {
         global.platform.closeCurrentWindow()
-      }, 2000)
-    } else if (getEnvironmentType() === ENVIRONMENT_TYPE_POPUP) {
+      }, 1500)
+    } else if (location.pathname === confirmPermissionPath) {
+      setTimeout(async () => {
+        history.push(DEFAULT_ROUTE)
+      }, 1500)
```

Comment:
Without this, the fullscreen does not show the redirect flow, and instead just sits on the account selection page until it disappears.